FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -28,15 +28,49 @@ public final class CoreDataFeedStore: FeedStore {
 		context = container.newBackgroundContext()
 	}
 
-	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+	func retrieve(completion: @escaping RetrievalCompletion) {
+		context.perform { [context] in
+			do {
+				guard let cache = try ManagedCache.find(in: context) else {
+					completion(.empty)
+					return
+				}
+				let feed = cache.feed.compactMap { ($0 as? ManagedFeedImage)?.local }
+				completion(.found(feed: feed, timestamp: cache.timestamp))
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
 
-	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+	func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		context.perform { [context] in
+			do {
+				let newCache = try ManagedCache.newUniqueInstance(in: context)
+				newCache.feed = ManagedFeedImage.images(from: feed, in: context)
+				newCache.timestamp = timestamp
+
+				try context.save()
+				completion(nil)
+			} catch {
+				context.rollback()
+				completion(error)
+			}
+		}
 	}
 
-	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
-		fatalError("Must be implemented")
+	func deleteCachedFeed(completion: @escaping DeletionCompletion) {
```

Comment:
This method should be `public` to be visible from outside of the module. Otherwise it would not be available for usage by module's clients.

Please make it `public`.

```suggestion
	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
```